### ISSUE TITLE
Release notes for 6.0

### ISSUE THREAD

**Issue by sheryjoe** [SCIInstitute/ShapeWorks#893]
Please add to this issue a paragraph (plus videos/snapshots for illustration) about each new/improved/bug-fix included in this release. I will consolidate and add to the documentation.

**Comment by akenmorris** [comment]
Added mesh support in Studio
Added support for loading feature maps from meshes

#848 - Studio: Changed default initial relative weighting
#856 - Studio: Fixed PCA mode setting to limit to number of samples - 2
#859 - Studio: Added ability to abort groom
#860 - Studio: Improved particle shape statistics computation speed
#870 - Studio: Add support for drag and drop of images and meshes
#880 - Studio: Fix but when adding and removing shapes
#892 - Studio: Ensure .xslx extension on Linux
#902 - Studio: Improve Optimize step responsiveness by loading images/meshes in worker thread
#801 - Studio: Added tooltips
#792 - Studio: Added Help->Keyboard shortcuts

#879 - Add new options to coverage command regarding back intersections

**Comment by sheryjoe** [comment]
Thanks @akenmorris! Could you add a "fancy" snapshot that showcase mesh and feature support in Studio?

**Comment by medakk** [comment]
* Support for use_normals in mesh domain: this results in improved shape models on thin domains:
![20201227_ellipsoid_withnormals](https://user-images.githubusercontent.com/10536745/103177587-1229d800-4839-11eb-93ed-222e6d869241.gif)
* Gradient of normals for image domain: this corrects a long standing bug where we used the hessian in place of the gradient of the normal. If you have an existing use case with use_normals enabled, the normal weighting may have to be adjusted
* Performance improvements:
    * ~20% improvement in all domains, from replacing certain data structures (that were originally in place to allow interactive removal of particles _during_ the optimization)
    * ~50% improvement in mesh domain, from caching nearest-triangle lookups

**Comment by sheryjoe** [comment]
@medakk thanks!
- could you add a snapshot for a couple of thin ellispsoid samples with their partices? the view should demonstrate the thin aspect of it.
- for performance improvement, i suppose this is memory improvement, right?

**Comment by medakk** [comment]
@sheryjoe 

> could you add a snapshot 

Sure!

> i suppose this is memory improvement, right?

No, its faster runtime. 

I updated my original comment

**Comment by jadie1** [comment]
DeepSSM:
- DeepSSM now saves both the "best" and "final" model. The final model is saved after all training epochs have run. The best model is saved after the epoch which had the lowest prediction error on the validation set. The best model makes use of early stopping to prevent overfitting. 
- The error meshes that are output from running the DeepSSM use case can now be visualized in Studio. These meshes have a distance scalar field on them which captures the distance between the true and predicted mesh. To view in studio simply run: `ShapeWorksStudio path/to/error/mesh.vtk`.

Data Augmentation:
- In data augmentation the user can now either specify how many PCA components to retain in embedding OR what percentage of population variability to retain.  For example, if the user specifies that 95% of population variability should be kept, then the number of components will be automatically selected such that less than 5% of shape variation is lost in embedding.
- Data augmentation can now be run on a dataset for which Procrustes was used in optimization.  When both the local and world .particle files are passed as arguments for data augmentation, the translation is accounted for in the augmented data. 
- A jupyter notebook which demonstrates the data augmentation process has been added (`ShapeWorks/DeepSSMViz/Examples/Python/notebooks/tutorials/getting-started-with-data-augmentation.ipynb`). In this notebook, parallel violin plots are used to visually compare the distribution of real and augmented data.

Use cases:
- All use cases now have a tiny test which can be run using the` --tiny_test` option. When the tiny test is run, only the data necessary for the test is downloaded rather than all of the data. 
- All of the use cases (mesh or segmentation based) can now be run on a subset of the data using the `--use_subsample` option. Note that the entire dataset is downloaded in this case so that a subset which is representative of the entire dataset can be selected. 
- `GroomUtils.py `now makes use of Python binding rather than calling command line tools.

Generating shape cohorts:
- Example shape cohorts can now be generated using the ShapeWorks package `GenerateShapeCohort`.  Currently cohorts of parameterized ellipsoids or supershapes can be generated. Options are available to specify the degree to which the cohort is groomed (i.e. a cohort can be generated to be in alignment or misaligned in various ways).
- A notebook was added (`ShapeWorks/DeepSSMViz/Examples/Python/notebooks/tutorials/getting-started-with-shape-cohort-generation.ipynb`) which demonstrates how to use `GenerateShapeCohort`.

**Comment by sheryjoe** [comment]
Please add/update this issue with release notes pertaining to issues you have resolved and merged to alpha/master.
@cchriste @akenmorris @archanasri @riddhishb @medakk @iyerkrithika21 @jadie1 @HeavenlyBerserker

**Comment by archanasri** [comment]
Updated ShapeWorks API: Consolidation of mesh-based grooming tools (smooth, decimate, invert normals, reflect, alignment, fill holes, probe volume at mesh vertices, clip, translate, scale, bounding box, quality control, surface to surface distance, to image and to distance transform). Addition of mesh-based query tools (center, center of mass, number of points, number of faces, get field names, set field, get field, set field value, get field value, get field range, get field mean, get field std and comparison). This includes a full compement of unit tests.

New Python ShapeWorks API: In addition to command line and C++ interface, there is Python interface for existing image-based, segmentation-based and mesh-based grooming tools and query tools. This includes a full component of unit tests.

**Comment by iyerkrithika21** [comment]
> 
> 
> DeepSSM:
> 
>     * DeepSSM now saves both the "best" and "final" model. The final model is saved after all training epochs have run. The best model is saved after the epoch which had the lowest prediction error on the validation set. The best model makes use of early stopping to prevent overfitting.
> 
>     * The error meshes that are output from running the DeepSSM use case can now be visualized in Studio. These meshes have a distance scalar field on them which captures the distance between the true and predicted mesh. To view in studio simply run: `ShapeWorksStudio path/to/error/mesh.vtk`.
> 
> 
> Data Augmentation:
> 
>     * In data augmentation the user can now either specify how many PCA components to retain in embedding OR what percentage of population variability to retain.  For example, if the user specifies that 95% of population variability should be kept, then the number of components will be automatically selected such that less than 5% of shape variation is lost in embedding.
> 
>     * Data augmentation can now be run on a dataset for which Procrustes was used in optimization.  When both the local and world .particle files are passed as arguments for data augmentation, the translation is accounted for in the augmented data.
> 
>     * A jupyter notebook which demonstrates the data augmentation process has been added (`ShapeWorks/DeepSSMViz/Examples/Python/notebooks/tutorials/getting-started-with-data-augmentation.ipynb`). In this notebook, parallel violin plots are used to visually compare the distribution of real and augmented data.
> 
> 
> Use cases:
> 
>     * All use cases now have a tiny test which can be run using the` --tiny_test` option. When the tiny test is run, only the data necessary for the test is downloaded rather than all of the data.
> 
>     * All of the use cases (mesh or segmentation based) can now be run on a subset of the data using the `--use_subsample` option. Note that the entire dataset is downloaded in this case so that a subset which is representative of the entire dataset can be selected.
> 
>     * `GroomUtils.py `now makes use of Python binding rather than calling command line tools.
> 
> 
> Generating shape cohorts:
> 
>     * Example shape cohorts can now be generated using the ShapeWorks package `GenerateShapeCohort`.  Currently cohorts of parameterized ellipsoids or supershapes can be generated. Options are available to specify the degree to which the cohort is groomed (i.e. a cohort can be generated to be in alignment or misaligned in various ways).
> 
>     * A notebook was added (`ShapeWorks/DeepSSMViz/Examples/Python/notebooks/tutorials/getting-started-with-shape-cohort-generation.ipynb`) which demonstrates how to use `GenerateShapeCohort`.

Videos and images for shape cohort generator:

![ellipsoid_segs](https://user-images.githubusercontent.com/11266258/108124136-110c6e80-7064-11eb-8828-f9bed3863e67.png)
![supershapes_Segs](https://user-images.githubusercontent.com/11266258/108124137-11a50500-7064-11eb-8125-411bf15c0751.png)
![supershaped_generate_shapes](https://user-images.githubusercontent.com/11266258/108124638-aad41b80-7064-11eb-8572-f80426888f15.gif)

![ellipsoid_generate_shapes](https://user-images.githubusercontent.com/11266258/108124520-8bd58980-7064-11eb-9027-6ff101a03473.gif)

**Comment by sheryjoe** [comment]
> • Added support for loading and displaying scalar values from feature map volumes
@akenmorris wasn't this already included in release 5.5? what is new in release 6?

**Comment by sheryjoe** [comment]
> DeepSSM:
> 
> * DeepSSM now saves both the "best" and "final" model. The final model is saved after all training epochs have run. The best model is saved after the epoch which had the lowest prediction error on the validation set. The best model makes use of early stopping to prevent overfitting.
> * The error meshes that are output from running the DeepSSM use case can now be visualized in Studio. These meshes have a distance scalar field on them which captures the distance between the true and predicted mesh. To view in studio simply run: `ShapeWorksStudio path/to/error/mesh.vtk`.
> 
> Data Augmentation:
> 
> * In data augmentation the user can now either specify how many PCA components to retain in embedding OR what percentage of population variability to retain.  For example, if the user specifies that 95% of population variability should be kept, then the number of components will be automatically selected such that less than 5% of shape variation is lost in embedding.
> * Data augmentation can now be run on a dataset for which Procrustes was used in optimization.  When both the local and world .particle files are passed as arguments for data augmentation, the translation is accounted for in the augmented data.
> * A jupyter notebook which demonstrates the data augmentation process has been added (`ShapeWorks/DeepSSMViz/Examples/Python/notebooks/tutorials/getting-started-with-data-augmentation.ipynb`). In this notebook, parallel violin plots are used to visually compare the distribution of real and augmented data.
> 

@jadie1 did you update documentation to reflect the above?

**Comment by akenmorris** [comment]
> > • Added support for loading and displaying scalar values from feature map volumes
> 
> @akenmorris wasn't this already included in release 5.5? what is new in release 6?

Yes, you're right, I think it got included because we didn't close the issue until just recently.  I will update the list above.

**Comment by sheryjoe** [comment]
@sheryjoe add fix in #1123 to the release notes.

**Comment by jadie1** [comment]
> > DeepSSM:
> > 
> > * DeepSSM now saves both the "best" and "final" model. The final model is saved after all training epochs have run. The best model is saved after the epoch which had the lowest prediction error on the validation set. The best model makes use of early stopping to prevent overfitting.
> > * The error meshes that are output from running the DeepSSM use case can now be visualized in Studio. These meshes have a distance scalar field on them which captures the distance between the true and predicted mesh. To view in studio simply run: `ShapeWorksStudio path/to/error/mesh.vtk`.
> > 
> > Data Augmentation:
> > 
> > * In data augmentation the user can now either specify how many PCA components to retain in embedding OR what percentage of population variability to retain.  For example, if the user specifies that 95% of population variability should be kept, then the number of components will be automatically selected such that less than 5% of shape variation is lost in embedding.
> > * Data augmentation can now be run on a dataset for which Procrustes was used in optimization.  When both the local and world .particle files are passed as arguments for data augmentation, the translation is accounted for in the augmented data.
> > * A jupyter notebook which demonstrates the data augmentation process has been added (`ShapeWorks/DeepSSMViz/Examples/Python/notebooks/tutorials/getting-started-with-data-augmentation.ipynb`). In this notebook, parallel violin plots are used to visually compare the distribution of real and augmented data.
> 
> @jadie1 did you update documentation to reflect the above?

@sheryjoe The documentation was updated for everything except the studio visualization. I just added that on the mkdocs branch: https://github.com/SCIInstitute/ShapeWorks/blob/mkdocs/docs/deep-learning/deep-ssm.md

**Comment by sheryjoe** [comment]
@akenmorris @medakk @jadie1 @iyerkrithika21 @HeavenlyBerserker @riddhishb 

What do you think of this as the graphical abstract for R6?

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/3253036/110414915-1b5fce00-804e-11eb-9496-fa9dac729783.png">

**Comment by medakk** [comment]
@sheryjoe This looks great! (I still have the same nitpick as last time, I'd prefer that the "ShapeWorks 6.0" text not obscure the logo)

**Comment by sheryjoe** [comment]
> @sheryjoe This looks great! (I still have the same nitpick as last time, I'd prefer that the "ShapeWorks 6.0" text not obscure the logo)

@akenmorris @medakk @jadie1 @iyerkrithika21 @HeavenlyBerserker @riddhishb

Whenever I move the logo or the text, symmetries are lost. How about this one? I am open to any suggestions here. 

<img width="683" alt="image" src="https://user-images.githubusercontent.com/3253036/110497725-f3a35100-80b3-11eb-957d-5a18af864737.png">

**Comment by akenmorris** [comment]
I think the earlier one was better.  The text/logo doesn't bother me the way it does @medakk   :)

**Comment by sheryjoe** [comment]
1:1 now :) ... any other voters? comments? suggestions?

**Comment by iyerkrithika21** [comment]
2:1  :-D
I felt the earlier version was better as the logo was visible. In the second version, the logo watermark is too faint in the background.

**Comment by jadie1** [comment]
I agree, 3:1

**Comment by archanasri** [comment]
3:2

**Comment by sheryjoe** [comment]
not a consensus yet, need more voters. @HeavenlyBerserker @riddhishb ?

**Comment by medakk** [comment]
>  I felt the earlier version was better as the logo was visible. In the second version, the logo watermark is too faint in the background.

I agree with this, I prefer the first one too. Maybe make the whole image a little bigger so we can fit logo+name separately
![image](https://user-images.githubusercontent.com/10536745/110524829-1f80ff80-80d1-11eb-8395-cdb27ecc2878.png)

**Comment by sheryjoe** [comment]
Final call for voters!! 

@akenmorris @archanasri @cchriste @medakk @jadie1 @iyerkrithika21 @HeavenlyBerserker @riddhishb
[A]

<img width="899" alt="image" src="https://user-images.githubusercontent.com/3253036/110657926-7c86bf00-817e-11eb-8d23-314daf8d7fc8.png">

veruss [B]
<img width="899" alt="image" src="https://user-images.githubusercontent.com/3253036/110657560-27e34400-817e-11eb-89e6-49afc7fbd600.png">

**Comment by akenmorris** [comment]
A

**Comment by jadie1** [comment]
A

**Comment by medakk** [comment]
B

**Comment by iyerkrithika21** [comment]
B

**Comment by archanasri** [comment]
B

**Comment by sheryjoe** [comment]
2:3 @HeavenlyBerserker @cchriste @riddhishb , please VOTE!!

**Comment by HeavenlyBerserker** [comment]
A

**Comment by riddhishb** [comment]
A

**Comment by sheryjoe** [comment]
Update release notes and abstract to de-emphasize python stuff.
See #1168

**Comment by sheryjoe** [comment]
done.
